src/function/table/sniff_csv.cpp

```
@@ -0,0 +1,222 @@
+				result->sample_size = NumericLimits<int64_t>::Maximum();
+			}
+		} else {
+			throw InvalidInputException("Invalid sniff_csv named parameter: %s . Only sample_size is accepted",
+			                            loption);
+		}
+	}
+	if (input.named_parameters.size() > 1) {
+		throw InvalidInputException(
+		    "Invalid number of named parameters for sniff_csv function, only sample_size is accepted");
+	}
+	// We want to return the whole CSV Configuration
+	// 1. Delimiter
+	return_types.emplace_back(LogicalType::VARCHAR);
+	names.emplace_back("Delimiter");
+	// 2. Quote
+	return_types.emplace_back(LogicalType::VARCHAR);
+	names.emplace_back("Quote");
+	// 3. Escape
+	return_types.emplace_back(LogicalType::VARCHAR);
+	names.emplace_back("Escape");
+	// 4. NewLine Delimiter
+	return_types.emplace_back(LogicalType::VARCHAR);
+	names.emplace_back("NewLine Delimiter");
+	// 5. Skip Rows
+	return_types.emplace_back(LogicalType::UINTEGER);
+	names.emplace_back("Skip Rows");
+	// 6. Has Header
+	return_types.emplace_back(LogicalType::BOOLEAN);
+	names.emplace_back("Has Header");
+	// 7. List<Struct<Column-Name:Types>>
+	return_types.emplace_back(LogicalType::VARCHAR);
+	names.emplace_back("Columns");
+	// 8. Date Format
+	return_types.emplace_back(LogicalType::VARCHAR);
+	names.emplace_back("Date Format");
+	// 9. Timestamp Format
+	return_types.emplace_back(LogicalType::VARCHAR);
+	names.emplace_back("Timestamp Format");
+	// 10. CSV read function with all the options used
+	return_types.emplace_back(LogicalType::VARCHAR);
+	names.emplace_back("Prompt");
+	return std::move(result);
+}
+
+string NewLineIdentifierToString(NewLineIdentifier identifier) {
+	switch (identifier) {
+	case NewLineIdentifier::SINGLE:
+		return "\\n";
+	case NewLineIdentifier::CARRY_ON:
+		return "\\r\\n";
+	case NewLineIdentifier::MIX:
+		return "mix";
+	case NewLineIdentifier::NOT_SET:
+		throw InternalException("NewLine Identifier must always be set after running the CSV sniffer");
```

nit: should we throw an exception on default here instead?